### PR TITLE
Issue 840

### PR DESCRIPTION
**Pull request by DailisLangovskis**
Need assist on remaining failing tests.
@jmacura  ->
 // this.HsLog.error(
    //   `Cannot decide a type of source of layer ${getTitle(layer)}`
    // ); 
This is the last remaining cause of test fails, because for some reason loading composition from test/data file, doesn't successfully find the layer type.

### CODE REVIEW

**Comment by jmacura** [comment]
Shall read
```
this.HsLog.warn(
    Cannot decide a type of source of layer ${getTitle(layer)}
);
```

It is not an error, it should only produce warning. My mistake, apologies!

**Comment by FilipLeitner** [comment]
> Shall read
> 
> ```
> this.HsLog.warn(
>     Cannot decide a type of source of layer ${getTitle(layer)}
> );
> ```
> 
> It is not an error, it should only produce warning. My mistake, apologies!

With error it fails beacuse of this function:
https://github.com/hslayers/hslayers-ng/blob/develop/projects/hslayers/src/components/utils/layer-utils.service.mock.ts#L7

We might want to mock it better

**Comment by DailisLangovskis** [comment]
@raitisbe The last fails happen from -> this.styleObject = await this.sldToJson(sld); on the third-party lib 
_stack: "TypeError: Cannot read property 'forEach' of undefined\n    at http://localhost:9876/_karma_webpack_/webpack:/projects/hslayers/node_modules/geostyler-sld-parser/build/dist/SldStyleParser.js:1090:1\n_
I am not sure, how to fix this, would appreciate some help.

**Comment by FilipLeitner** [comment]
> @raitisbe The last failing tests arises from -> this.styleObject = await this.sldToJson(sld); on the third-party lib
> _stack: "TypeError: Cannot read property 'forEach' of undefined\n at [http://localhost:9876/_karma_webpack_/webpack:/projects/hslayers/node_modules/geostyler-sld-parser/build/dist/SldStyleParser.js:1090:1\n](http://localhost:9876/_karma_webpack_/webpack:/projects/hslayers/node_modules/geostyler-sld-parser/build/dist/SldStyleParser.js:1090:1%5Cn)_
> I am not sure, how to fix this, would appreciate some help.

Tracked the error down to be cause by this [function ](https://github.com/hslayers/hslayers-ng/blob/develop/projects/hslayers/src/components/styles/styler.spec.ts#L104)call ..in some specific occasions. or rather when styler tests were run in some specific sequence. Havent got exacty to that 'forEach' call but when test were not run in order it failed and Vector layerd had sld already defined (seemed like sld from 'SLD should be generated from OL style' test ...so it might be that test runnig before caused that)
...so it was basically either to define vector layer before each test or run tests in order (which I tought might be better ...in general)

**Comment by raitisbe** [comment]
Or could clear/reset the layer data in the failing test

File: projects/hslayers/src/testing-utils.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const orgConsoleError = window.console.error;
+
+export function patchConsoleToFailOnError(): void {
```
Comment (raitisbe) [inline projects/hslayers/src/testing-utils.ts:3]: Nice one

**Review by raitisbe — APPROVED** [review]

**Pull request merged**